apps/web/app/[locale]/(app)/dashboard/[organization]/[list]/import-contacts.tsx

```diff
@@ -0,0 +1,190 @@
+'use client';
+
+import { Alert, AlertTitle, AlertDescription } from '@/components/ui/alert';
+import { Button } from '@/components/ui/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import { ExclamationTriangleIcon } from '@radix-ui/react-icons';
+import { PlusIcon } from 'lucide-react';
+import { useDropzone } from 'react-dropzone';
+import Papa, { ParseResult } from 'papaparse';
+import { LinkedInImportContact } from '@/lib/types/import-contacts';
+import { useState } from 'react';
+import { useToast } from '@/components/ui/use-toast';
+import { useRouter } from 'next/navigation';
+import Link from 'next/link';
+
+interface ImportContactsModalProps {
+  organization: string;
+  list: string;
+}
+
+export const ImportContactsModal = ({
+  organization,
+  list,
+}: ImportContactsModalProps) => {
+  const [importedContacts, setImportedContacts] = useState<
+    LinkedInImportContact[]
+  >([]);
+  const [loading, setLoading] = useState(false);
+  const [modalOpen, setModalOpen] = useState(false);
+  const { toast } = useToast();
+  const router = useRouter();
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    accept: {
+      'text/csv': [],
+    },
+    onDropAccepted: acceptedFiles => {
+      const file = acceptedFiles[0];
+      if (!file) return;
+      const reader = new FileReader();
+      reader.onabort = () => console.log('file reading was aborted');
+      reader.onerror = () => console.log('file reading has failed');
```

Comment:
Not sure if and how the libraries handle errors, but it would be nice to have an error state for the UI - I suspect that such imports are messed up easily 😄

Reply:
Yep, good one. Added some sentry handler and toasts for giving feedback to the user in case an error occurs.